### PR TITLE
Handle ActorInitializationException from Classic child in Typed parent, #27970

### PR DESCRIPTION
**Pull request by patriknw**
* otherwise restart loop

Refs #27970

Reproduced this with ClusterReceptionist as reported in the issue, then created minimized test.

### CODE REVIEW

File: akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorAdapter.scala [hunk 1]
```diff
@@ -223,8 +223,10 @@ import akka.util.OptionVal
       log.error(ex, logMessage)
       if (isTypedActor)
         classic.SupervisorStrategy.Stop
-      else
-        classic.SupervisorStrategy.Restart
+      else {
+        // ActorInitializationException => Stop in defaultDecider
+        classic.SupervisorStrategy.defaultDecider.applyOrElse(ex, (_: Throwable) => classic.SupervisorStrategy.Restart)
```
Comment (johanandren) [inline akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorAdapter.scala]: Should we cache the function, or even the entire decider?

**Review by johanandren — APPROVED** [review]
LGTM

File: akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorAdapter.scala (hunk 1 above)
Comment (patriknw) [reply akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorAdapter.scala]: ok: https://github.com/akka/akka/pull/27991/commits/fceca136b98e13aa58d6667a9d9cb415bf5c05c4

**Review by johanandren — APPROVED** [review]

**Review by chbatey — APPROVED** [review]
LGTM

**Pull request merged**